### DISCUSSION TITLE
Is there a limit on the number of Procedures to display?

### DISCUSSION THREAD

**Discussion by smhubert** [dbeaver/dbeaver#16920] [General]
I'm using Firebird 2.5.4 and when I view the stored procedures in DBeaver it only shows the procedures A-D.  I know there are more procedures in my database.  How can I display them all?

![image](https://user-images.githubusercontent.com/99685712/176508120-73bd89e1-6d46-4e52-9ec9-6c167e7a5ac2.png)

**Comment by LonwoLonwo** [comment]
What DBeaver version do you use? And what FireBird driver version do you use?

**Comment by smhubert** [comment]
Hi Anastasiya,
Sorry, I should have included that in my original question.
I'm using DBeaver Version 22.1.2.202207091909 (upgraded today and it still
has the same issue)
and Firebird Jaybird driver version
[image: image.png]

Thank you

**Reply by LonwoLonwo** [reply]
Can you please update image here too? About driver version. Did you try to change driver version? In Connection settings -> Edit driver settings ->Libraries

**Comment by LonwoLonwo** [comment]
![2022-07-11 17_42_27-DBeaver 21 1 0 - _employee fdb_ Script-64](https://user-images.githubusercontent.com/45152336/178290872-759a9df4-74c5-4d72-8657-05fd20d153fb.png)

I see on your screen Procedures and Functions in separate folders. Did you enable this setting or use another connection for Firebird - not standard?

**Comment by smhubert** [comment]
Yes, I did enable this setting to see if it would fix the issue.

**Comment by LonwoLonwo** [comment]
I can't reproduce your issue for now.

![2022-07-11 17_51_48-DBeaver Ultimate 22 1 2 - _postgres12local_ Script-170](https://user-images.githubusercontent.com/45152336/178293003-5d18b698-0de8-4818-990a-1e69f8e4ce2c.png)

Maybe an advanced connection view can help?

![2022-07-11 17_52_53-DBeaver Ultimate 22 1 2 - _postgres12local_ Script-170](https://user-images.githubusercontent.com/45152336/178293162-bcada972-15ae-41ff-b2d7-2f73f7a780f9.png)

If not, can you please specify your driver version? You can check it in the Connection Settings - Test Connection.

**Comment by smhubert** [comment]
Hi Anastasiya,
I found something that might help.
I know of a procedure that is not in the procedure list.  When typing in
the procedure name, the look ahead in DBeaver successfully showed me the
procedure name, and other procedures that matched (auto complete).  None of
these show up in the Procedure tree in Navigation.
Maybe a clue to what is going on is the DBException that is thrown about
not being able to find the procedure (in yellow).
![image](https://user-images.githubusercontent.com/99685712/180016444-25ad8b6c-2b92-4f95-b207-7574c42a667b.png)



I can successfully execute the procedure, despite DBeaver saying it
couldn't find it.
![image (1)](https://user-images.githubusercontent.com/99685712/180016565-5c336950-b198-4e5c-acfd-c189a2e8399b.png)


Hope this helps!

Thanks

**Reply by LonwoLonwo** [reply]
Hello @smhubert 

Can you please provide the image on this GitHub page? You can't send it via e-mail.

**Reply by smhubert** [reply]
Ok @LonwoLonwo, I updated my previous post.

**Comment by smhubert** [comment]
Server: Firebird 2.5 WI-V2.5.4.26856
Driver: Jaybird JCA/JDBC driver 4.0.0
![image](https://user-images.githubusercontent.com/99685712/180286756-591ddead-b27c-4ec2-b5a4-aa7a32c97606.png)

and libraries:
![image](https://user-images.githubusercontent.com/99685712/180286916-d4b7aced-582c-4b93-8e2b-9fe30c11c20b.png)

**Comment by LonwoLonwo** [comment]
Can you try to update your driver version, please? Maybe, it helps.

![2022-07-21 21_48_06-Connection _sample fdb_ configuration](https://user-images.githubusercontent.com/45152336/180293106-5efee66f-9c7c-4214-8da0-bee98d96b50d.png)

![2022-07-21 21_48_31-DBeaver 22 1 2 - _sqlite_cript db_ Script-34](https://user-images.githubusercontent.com/45152336/180293113-c99d1dce-8928-4653-a1d0-9474af7723ff.png)

**Reply by smhubert** [reply]
@LonwoLonwo I was so hopeful, but it didn't change anything.  I went to version 4.0.6.java11

![image](https://user-images.githubusercontent.com/99685712/180476114-3f470439-0826-40df-96f8-dbe77f852c86.png)

**Comment by smhubert** [comment]
Thanks for all your help, but I had to try another program (Red Expert) and it worked.  I can now see all 236 Procedures instead of just the 54 in DBeaver.  It is a shame because I like the DBeaver user interface better.

![image](https://user-images.githubusercontent.com/99685712/180571654-69c07d37-1b20-42fa-b515-6370ef85942e.png)

**Reply by LonwoLonwo** [reply]
Give me the last chance, please.

Can you please execute this query
`SELECT * FROM RDB$PROCEDURES`
Or check the RDB$PROCEDURES table in the database navigator (which will be available with the Advanced navigator view)?
Does this table contain the info about your procedures?

**Comment by smhubert** [comment]
Yes!  All 236 procedures show up with both the query and the RDB$PROCEDURES table.
But it is difficult to look at the source when viewing the procedures like this.  I have to go to the RDB$PROCEDURE_SOURCE column and view it with minimal formatting.

**Reply by LonwoLonwo** [reply]
I created a new ticket about reading procedures from system tables, not from the driver. 
https://github.com/dbeaver/dbeaver/issues/17169
In this case, the procedures list will be shown correctly.

**Comment by smhubert** [comment]
Hi Anastasiya,
Thank you for the excellent support.
I tried the Advanced connection view and it didn't resolve the issue.
My driver version is: Jaybird JCA/JDBC driver 4.0.0

Thank you

**Comment by tonne1** [comment]
Although this is an old ticket, this has been resolved in DBeaver or was a driver issue. I have a Firebird database with about 690 stored procedures and all of them are listed in the navigator tree. Using Firebird 4 and 5, but should apply to older versions too. The JDBC driver makes a difference, I'm using org.firebirdsql.jdbc:jaybird:RELEASE, Version 5.0.4.java11.

**Reply by LonwoLonwo** [reply]
Thanks for the clarification @tonne1
